opencti-platform/opencti-front/src/private/components/dataEntities/data/Roles/EntitiesRolesDeletion.js

```diff
@@ -0,0 +1,231 @@
+import Tooltip from '@material-ui/core/Tooltip';
+import graphql from 'babel-plugin-relay/macro';
+import { QueryRenderer as QR, commitMutation as CM } from 'react-relay';
+import inject18n from '../../../../../components/i18n';
+import environmentDarkLight from '../../../../../relay/environmentDarkLight';
+import { commitMutation, QueryRenderer } from '../../../../../relay/environment';
+import Loader from '../../../../../components/Loader';
+import Security, {
+  KNOWLEDGE_KNUPDATE,
+  KNOWLEDGE_KNUPDATE_KNDELETE,
+} from '../../../../../utils/Security';
+import { toastGenericError } from '../../../../../utils/bakedToast';
+
+const styles = (theme) => ({
+  container: {
+    margin: 0,
+  },
+  iconButton: {
+    float: 'left',
+    minWidth: '0px',
+    marginRight: 15,
+    padding: '7px',
+  },
+  drawerPaper: {
+    minHeight: '100vh',
+    width: '50%',
+    position: 'fixed',
+    overflow: 'auto',
+    backgroundColor: theme.palette.navAlt.background,
+    transition: theme.transitions.create('width', {
+      easing: theme.transitions.easing.sharp,
+      duration: theme.transitions.duration.enteringScreen,
+    }),
+    padding: 0,
+  },
+  buttonPopover: {
+    textTransform: 'capitalize',
+  },
+  dialogActions: {
+    justifyContent: 'flex-start',
+    padding: '10px 0 20px 22px',
+  },
+});
+
+const Transition = React.forwardRef((props, ref) => (
+  <Slide direction="up" ref={ref} {...props} />
+));
+Transition.displayName = 'TransitionSlide';
+
+const EntitiesRolesDeletionMutation = graphql`
+  mutation EntitiesRolesDeletionMutation($id: ID!) {
+    threatActorEdit(id: $id) {
+      delete
+    }
+  }
+`;
+
+const EntitiesRolesDeletionDarkLightMutation = graphql`
+  mutation EntitiesRolesDeletionDarkLightMutation($id: ID!) {
+  deleteComputingDeviceAsset(id: $id)
+}
+`;
+
+class EntitiesRolesDeletion extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      anchorEl: null,
+      displayDelete: false,
+      displayEdit: false,
+      deleting: false,
+    };
+  }
+
+  handleOpen(event) {
+    this.setState({ anchorEl: event.currentTarget });
+  }
+
+  handleClose() {
+    this.setState({ anchorEl: null });
+  }
+
+  handleOpenDelete() {
+    this.setState({ displayDelete: true });
+    this.handleClose();
+  }
+
+  handleCloseDelete() {
+    this.setState({ displayDelete: false });
+  }
+
+  handleOpenEdit() {
+    this.setState({ displayEdit: true });
+    this.handleClose();
+  }
+
+  handleCloseEdit() {
+    this.setState({ displayEdit: false });
+  }
+
+  submitDelete() {
+    this.setState({ deleting: true });
+    CM(environmentDarkLight, {
+      mutation: EntitiesRolesDeletionDarkLightMutation,
+      variables: {
+        id: this.props.id,
+      },
+      onCompleted: (data) => {
+        this.setState({ deleting: false });
+        this.handleClose();
+      },
+      onError: (err) => {
+        console.error(err);
+        return toastGenericError('Failed to delete role');
```

Comment:
@sachin755 there is nothing to return from toasts, please remove return.

Reply:
@sachin755 disregard. I will push with those changes. But please note to not return from the toast call